test/parallel/test-shadow-realm-allowed-builtin-modules.js

```diff
@@ -0,0 +1,37 @@
+// Flags: --experimental-shadow-realm
+'use strict';
+const common = require('../common');
+const fixtures = require('../common/fixtures');
+const assert = require('assert');
+const { builtinModules } = require('module');
+
+async function main() {
+  const realm = new ShadowRealm();
+  const mod = fixtures.fileURL('es-module-shadow-realm', 'builtin-modules.mjs');
+  const builtinModulesStr = await realm.importValue(mod, 'builtinModulesStr');
+  const shadowRealmBuiltinModules = builtinModulesStr.split(',');
+  const diff = [];
+  for (const mod of builtinModules) {
+    if (!shadowRealmBuiltinModules.includes(mod)) {
+      diff.push(mod);
+    }
+  }
+  // Check that not all builtin modules are available in the ShadowRealm.
+  assert.ok(diff.length > 0);
+
+  // Verify that dynamic import can not import disallowed builtin modules.
+  assert.ok(!shadowRealmBuiltinModules.includes('fs'));
+  await assert.rejects(realm.importValue('fs', 'readFileSync'), {
+    message: /Cannot find package 'fs'/,
```

Comment:
Should we check the error code instead?

Reply:
Do you want to add a comment explaining that?

Reply:
Yeah, thank you for the suggestion!